lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -477,6 +495,19 @@ def _fetch_local_cm_nodes(self, deployment_desc: DeploymentDescription) -> List[
             use_localhost=self._opensearch.is_node_up(),
             hosts=self.charm.alt_hosts,
         )
+
+        if not nodes:
+            # create a node from the deployment desc and unit data only
+            return [
+                Node(
+                    name=self.charm.unit_name,
+                    roles=deployment_desc.config.roles,
```

Comment:
We need to handle the case where the roles are autogenerated, this list will be None

Reply:
done